content/topics/containers.md

```diff
@@ -1,43 +1,198 @@
+              image: "nginx",
+              ports: [{ port: 80 }],
+              replicas: 2,
+          });
+
+          export let url = nginx.defaultEndpoint;
+      cta:
+          url: /quickstart/aws-containers
+          label: GET STARTED
+
+    - title: Deploying with a custom build
+      body: >
+          This example uses a trivial Dockerfile that derives from the <code>nginx</code> base image and copies the
+          <code>./www</code> directory into the nginx HTML target so that it will be served up.
+      code: |
+          // Using a custom build based on Nginx
+          import * as cloud from "@pulumi/cloud";
+
+          let nginx = new cloud.Service("nginx", {
+              build: ".",
+              ports: [{ port: 80 }],
+              replicas: 2,
+          });
+
+          export let url = nginx.defaultEndpoint;
+
+          // Dockerfile
+          FROM nginx
+          COPY ./www /usr/share/nginx/html
+      cta:
+          url: /quickstart/aws-containers
+          label: GET STARTED
+
+    - title: Connecting containers
+      body: >
+          This example shows how we can connect containers using Pulumi &mdash; in this case Redis for a data store,
+          and a Python flask app for a front end. Using Pulumi, it is easy to obtain a reference to the container
+          objects, and connect them using code.
+      code: |
+          import * as pulumi from "@pulumi/pulumi";
+          import * as cloud from "@pulumi/cloud";
+
+          // The data layer for the application
+          let redisCache = new cloud.Service("voting-app-cache", {
+              containers: {
+                  redis: {
+                      image: "redis:alpine",
+                      memory: 512,
+                      ports: [{ port: redisPort }],
+                      command: ["redis-server", "--requirepass", redisPassword],
+                  },
+              },
+          });
+
+          let redisEndpoint = redisCache.endpoints.apply(endpoints => endpoints.redis[redisPort]);
+
+          // A custom container for the frontend, which is a Python Flask app
+          let frontend = new cloud.Service("voting-app-frontend", {
+              containers: {
+                  votingAppFrontend: {
+                      build: "./frontend",   // path to Dockerfile folder
+                      memory: 512,
+                      ports: [{ port: 80 }],
+                      environment: {
+                          // pass the Redis container info in environment variables
+                          "REDIS":      redisEndpoint.apply(e => e.hostname),
+                          "REDIS_PORT": redisEndpoint.apply(e => e.port.toString()),
+                          "REDIS_PWD":  redisPassword
+                      }
+                  },
+              },
+          });
+
+          export let frontendURL = frontend.endpoints.apply(e => e["votingAppFrontend"][80].hostname);
+      cta:
+          url: /quickstart/aws-containers
```

Comment:
This appears to be a broken URL, from looking at the Travis build for this PR.

Reply:
Ah, yes -- appears to have been relocated to `/docs/quickstart/cloudfx/tutorial-service`. Fixed up.